lib/rules.rb

```diff
@@ -0,0 +1,99 @@
+class Rules 
+  
+  def valid_marker?(player_marker_selection)
+    ["X", "x", "O", "o"].each do |acceptable_marker|
+      return true if player_marker_selection == acceptable_marker
+    end
+    false
+  end
+   
+  def valid_move?(board, user_selection)
+    if valid_move_input?(user_selection)
+      return true if box_is_empty?(board, user_selection)
+    end
+    false
+  end
+
+  def game_won?(board)
+    potentially_winning_paths = get_potentially_winning_paths(board)
+    potentially_winning_paths.each do |path|
+      return true if has_win?(path)
+    end
+    false
+  end
+
+  def player_won_game?(board, player)
+    potentially_winning_paths = get_potentially_winning_paths(board)
+    potentially_winning_paths.each do |path|
+      return true if has_win?(path) && path[0] == player.marker
+    end
+    false
+  end
```

Comment:
Code Smell: Duplicative Code/Uncommunicative Names.  How are these two methods different exactly? Why are they both needed? Can they be combined somehow?